config/infrastructure/rbac/kubernetes/builder-role-binding-kubernetes.yaml

```diff
@@ -0,0 +1,30 @@
+# ---------------------------------------------------------------------------
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ---------------------------------------------------------------------------
+
+kind: RoleBinding
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: camel-k-builder-kubernetes
+  labels:
```

Comment:
The suffixes of "-openshift" and "-kubernetes" became necessary from a practical step that 2 bindings cannot have the same name. If kustomize encounters 2 resources of the same kind with the same name it results in an error. In some cases, the openshift and kubernetes resources are mutually exclusive and thus only 1 or the other is required to be applied. However, in other cases, the openshift version simply adds extra permissions to the 'base' kubernetes version. Therefore,
- need to decide if openshift resources should duplicate kubernetes content and so only 1 or the other is installed. This would make suffixing resources unnecessary;
- if openshift resources provide extra openshift-specific content only and so require the kubernetes resource as a base. This would require the suffixing of resource names.

My personal preference is the latter as I am not a fan of duplication and I think making it obvious to the user that such permissions and bindings affect only a specific cluster type is transparent and informative.

Reply:
My comment here was about labels, and was proposing to use `commonLabels` instead of duplicating labels in resources, which can be error-prone.

Reply:
For the suffixing, I agree we should avoid duplication as much as possible. From what I see, it seems `camel-k-builder-kubernetes` and `camel-k-builder-openshift` Roles have duplicated rules, while being suffixed.

Reply:
Updated these roles to be additve rather than duplicating so renamed to `camel-k-builder-core` and `camel-k-builder-openshift`.